sdk/v2/core/projects_test.go

```diff
@@ -254,3 +254,57 @@ func TestProjectsClientDelete(t *testing.T) {
 	err := client.Delete(context.Background(), testProjectID)
 	require.NoError(t, err)
 }
+
+func TestValidateProjectID(t *testing.T) {
+	testCases := []struct {
+		name     string
+		id       string
+		expected bool
+	}{
+		{
+			name:     "Valid id with minimum amount of allowed characters",
+			id:       "abc",
+			expected: true,
+		},
+		{
+			name: "Valid id with minimum amount of" +
+				" allowed characters and numbers",
+			id:       "a11",
+			expected: true,
+		},
+		{
+			name: "Valid id with maximum amount of allowed characters",
+			id: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" +
+				"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+			expected: true,
+		},
```

Comment:
Let's collapse these into just once success case. It's interesting and useful to test lots of failure cases-- make validation fail for different reasons-- but it's less interesting to test different success cases. There's _really_ only one success case-- you succeed when you don't fail.